homeassistant/components/fan/wemo.py

```diff
@@ -0,0 +1,248 @@
+        return self.wemo.name
+
+    @property
+    def is_on(self):
+        """Return true if switch is on. Standby is on."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Return the icon of device based on its type."""
+        return 'mdi:water-percent'
+
+    def turn_on(self: ToggleEntity, speed: str=None, **kwargs) -> None:
+        """Turn the switch on."""
+        self._state = WEMO_ON
+
+        if speed is None:
+            self.wemo.set_state(self._last_fan_on_mode)
+        else:
+            self.set_speed(speed)
+
+        self.schedule_update_ha_state()
+
+    def turn_off(self: ToggleEntity, **kwargs) -> None:
+        """Turn the switch off."""
+        self._state = WEMO_OFF
+        self.wemo.set_state(WEMO_FAN_OFF)
+
+        self.schedule_update_ha_state()
+
+    def set_speed(self: ToggleEntity, speed: str) -> None:
+        """Set the fan_mode of the Humidifier."""
+        if speed == SPEED_OFF:
+            self.wemo.set_state(WEMO_FAN_OFF)
+            self._state = WEMO_OFF
+        elif speed == SPEED_MINIMUM:
+            self.wemo.set_state(WEMO_FAN_MINIMUM)
+            self._state = WEMO_ON
+        elif speed == SPEED_LOW:
+            self.wemo.set_state(WEMO_FAN_LOW)
+            self._state = WEMO_ON
+        elif speed == SPEED_MEDIUM:
+            self.wemo.set_state(WEMO_FAN_MEDIUM)
+            self._state = WEMO_ON
+        elif speed == SPEED_HIGH:
+            self.wemo.set_state(WEMO_FAN_HIGH)
+            self._state = WEMO_ON
+        elif speed == SPEED_MAXIMUM:
+            self.wemo.set_state(WEMO_FAN_MAXIMUM)
+            self._state = WEMO_ON
+
+        self.schedule_update_ha_state()
+
+    def set_humidity(self: ToggleEntity, humidity: float) -> None:
+        """Set the target humidity level for the Humidifier."""
+        if humidity < 50:
+            self.wemo.set_humidity(WEMO_HUMIDITY_45)
+        elif humidity >= 50 and humidity < 55:
+            self.wemo.set_humidity(WEMO_HUMIDITY_50)
+        elif humidity >= 55 and humidity < 60:
+            self.wemo.set_humidity(WEMO_HUMIDITY_55)
+        elif humidity >= 60 and humidity < 100:
+            self.wemo.set_humidity(WEMO_HUMIDITY_60)
+        elif humidity >= 100:
+            self.wemo.set_humidity(WEMO_HUMIDITY_100)
+
+        self.schedule_update_ha_state()
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            "current_humidity": self._current_humidity,
+            "target_humidity": self._target_humidity,
+            "fan_mode": self._fan_mode,
+            "water_level": self._water_level,
+            "filter_life": self._filter_life,
+            "filter_expired": self._filter_expired
+        }
+
+    @property
+    def speed(self) -> str:
+        """Return the current speed."""
+        current_wemo_speed = self.wemo.fan_mode
+        if WEMO_FAN_OFF == current_wemo_speed:
```

Comment:
Instead of all the `if ... elif`, make a dict that maps the wemo state to the home assistant state. Define the dict at the module level above. Then lookup that dict here using the wemo speed state, and return the found home assistant state.